worker-workflow/src/main/java/com/github/cafdataprocessing/workflow/WorkflowWorker.java

```diff
@@ -143,4 +149,35 @@ public void processDocument(final Document document) throws DocumentWorkerTransi
             document.addFailure("WORKFLOW_SCRIPT_EXCEPTION", e.getMessage());
         }
     }
+    
+    private void addMdcLoggingData(final Document document)
+    {
+        // The logging pattern we use uses a tenantId and a correlationId:
+        // 
+        // https://github.com/CAFapi/caf-logging/tree/4ef35ae3a6da4329a427667782f7aaff4fee8c1d#pattern
+        // https://github.com/CAFapi/caf-logging/blob/4ef35ae3a6da4329a427667782f7aaff4fee8c1d/src/main/resources/logback.xml#L27
+        //
+        // This function adds a tenantId and correlationID to the MDC (http://logback.qos.ch/manual/mdc.html), so that log messages from 
+        // *this* worker (workflow-worker) will contain these values.
+        //
+        // See also addMdcData in workflow-control.js, which performs similar logic to ensure log messages from *subsequent* workers in 
+        // the workflow also contain these values. 
+
+        // Get MDC data from custom data.
+        final String tenantId = document.getTask().getCustomData(TENANT_ID_KEY);
+        String correlationId = document.getTask().getCustomData(CORRELATION_ID_KEY);
```

Comment:
When you can't use `final`, it if often an indication that additional functional decomposition would help (which is why I always encourage the use of `final`).  For example in this case, consider introducing a utility function to get the correlation id:

    private static String getOrCreateCorrelationId(final Task task)
    {
        final String correlationId = task.getCustomData(CORRELATION_ID_KEY);

        return (correlationId == null)
            ? UUID.randomUUID().toString()
            : correlationId;
    }